eslint.js

```diff
@@ -34,88 +34,34 @@ const eslintAnnotations = async (
-    // $FlowFixMe: flow can't handle custom requires
-    const eslint = require(path.resolve(eslintDirectory));
-
-    let results /*: Array<LintResult> */ = [];
-    let formatter /*: Formatter */;
-
-    if (eslint.ESLint) {
-        core.info(`version: ${eslint.ESLint.version}`);
-        const cli = new eslint.ESLint({
-            resolvePluginsRelativeTo: baseDirectory,
-        });
-        formatter = await cli.loadFormatter('stylish');
-        results = await cli.lintFiles(files);
-    } else if (eslint.CLIEngine) {
-        // Handle old versions of eslint (< 7)
-        core.info(`version: ${eslint.CLIEngine.version}`);
-        const cli = new eslint.CLIEngine({
-            resolvePluginsRelativeTo: baseDirectory,
-        });
-        formatter = {
-            format: cli.getFormatter('stylish'),
-        };
-        const report /*: LintReport */ = cli.executeOnFiles(files);
-        /* end flow-uncovered-block */
-        results = report.results;
-    } else {
-        throw new Error(`'eslint-lib: ${eslintDirectory}' is incorrect`);
-    }
-
     /* flow-uncovered-block */
     // Log which files are being linted.
     const cwd = process.cwd();
-    core.startGroup('Running eslint on the following files:');
-    for (const file of files) {
-        core.info(path.relative(cwd, file));
-    }
-    core.endGroup();
-
-    // Log all results since the number of annotations we can have is limited.
-    core.startGroup('Results:');
-    core.info(formatter.format(results));
-    core.endGroup();
-    /* end flow-uncovered-block */
-
-    const annotations /*: Array<Message> */ = [];
-    for (const result of results) {
-        const {filePath, messages} = result;
-        for (const msg of messages) {
-            const {line, column, severity, ruleId, message} = msg;
-            if (!ruleId || severity === 0) {
-                // it's probably the warning about a given file being ignored
-                // by .eslintignore, which is fine.
-                continue;
-            }
-            annotations.push({
-                path: filePath,
-                start: {line, column},
-                end: {line, column},
-                annotationLevel: severity === 1 ? 'warning' : 'failure',
-                message: `${chalk.red(`[${ruleId}]`)} ${message}`,
-            });
+    if (files.length === 1 && files[0] === '.') {
+        core.info(`Linting all relevant files in ${cwd}`);
+    } else {
+        core.startGroup('Running eslint on the following files:');
+        for (const file of files) {
+            core.info(path.relative(cwd, file));
         }
+        core.endGroup();
     }
 
-    return annotations;
-};
+    const args = [
+      path.resolve(eslintDirectory, 'bin', 'eslint'),
+      ...files,
+    ].filter(Boolean);
 
-const parseList = (text) /*: Array<string>*/ => {
-    if (!text || !text.length) {
-        return [];
-    }
-    return text.split(',');
+    return await exec('node', args);
```

Comment:
Currently relying on auto config discovery. Could add a `config` input if we really wanted, so that we could override this, but our old version didn't bother.